packages/typespec-client-generator-core/test/decorators.test.ts

```diff
@@ -2197,4 +2206,102 @@ describe("typespec-client-generator-core: decorators", () => {
       strictEqual(getUsage(runner.context, Dog), UsageFlags.Output);
     });
   });
+
+  describe("@flattenProperty", () => {
+    it("marks a model property to be flattened with suppression of deprecation warning", async () => {
+      const { Model1 } = (await runner.compile(`
+        @service({})
+        @test namespace MyService {
+          @test
+          model Model1{
+            #suppress "deprecated" "@flattenProperty decorator is not recommended to use."
+            @flattenProperty
+            child: Model2;
+          }
+
+          @test
+          model Model2{}
+
+          @test
+          @route("/func1")
+          op func1(@body body: Model1): void;
+        }
+      `)) as { Model1: Model };
+
+      const childProperty = Model1.properties.get("child");
+      notEqual(childProperty, undefined);
+      strictEqual(shouldFlattenProperty(runner.context, childProperty as ModelProperty), true);
+    });
+
+    it("doesn't mark a un-flattened model property", async () => {
+      const { Model1 } = (await runner.compile(`
+        @service({})
+        @test namespace MyService {
+          @test
+          model Model1{
+            child: Model2;
+          }
+
+          @test
+          model Model2{}
+
+          @test
+          @route("/func1")
+          op func1(@body body: Model1): void;
+        }
+      `)) as { Model1: Model };
+
+      const childProperty = Model1.properties.get("child");
+      notEqual(childProperty, undefined);
```

Comment:
```suggestion
      strictNotEqual(childProperty, undefined);
```
or use the new `vitest` `expect`

Reply:
replaced with `notStrictEqual`